libra-node/src/main_node.rs

```diff
@@ -244,17 +190,32 @@ pub fn setup_environment(node_config: &mut NodeConfig) -> (AdmissionControlClien
     debug!("Executor setup in {} ms", instant.elapsed().as_millis());
     let mut network_runtimes = vec![];
     let mut state_sync_network_handles = vec![];
+    let mut ac_network_sender = None;
+    let mut ac_network_events = vec![];
     let mut validator_network_provider = None;
 
-    for mut network in &mut node_config.networks {
-        let peer_id = PeerId::try_from(network.peer_id.clone()).expect("Invalid PeerId");
-        let (runtime, mut network_provider) = setup_network(peer_id, &mut network);
+    for i in 0..node_config.networks.len() {
```

Comment:
nit: why you need to change the way you iterate?
also you could use enumerate instead

Reply:
Because currently it iterates the networks mutably (&mut node_config.networks), so we can't make any other calls that borrow node_config inside the loop. Trying to isolate the mutably borrowed scope to just setup_network by doing &mut node_config.networks[i] instead